### PR TITLE
Cosmos DB - Re organize files into data

### PR DESCRIPTION
**Pull request by ealsur**
Move `sdk/cosmos` to `sdk/data/azcosmos`

### CODE REVIEW

**Comment by weshaggard** [comment]
Moving the ci.yml file is going to break the pipeline so that will need to be updated once this change is ready/in.

**Review by ausfeldt — APPROVED** [review]

**Comment by benbp** [comment]
@ausfeldt ping me when you merge I can update the internal+public pipelines.

**Comment by benbp** [comment]
I think to be consistent this should be `data/azcosmos` not `data/cosmos/azcosmos`?

**Comment by ealsur** [comment]
@benbp Based on https://azure.github.io/azure-sdk/golang_introduction.html#golang-packages, there could be an `armcosmos` package also for management operations, so the folder makes sense to group both up? @jhendrixMSFT Could you confirm?

File: sdk/data/cosmos/azcosmos/go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module github.com/Azure/azure-sdk-for-go/sdk/cosmos/azcosmos
+module github.com/Azure/azure-sdk-for-go/sdk/data/cosmos/azcosmos
 
 go 1.13
```
Comment (seankane-msft) [inline sdk/data/cosmos/azcosmos/go.mod:3]: This should be go 1.16, which is our new min supported version

File: sdk/data/cosmos/azcosmos_emulator_tests/go.mod [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module github.com/Azure/azure-sdk-for-go/sdk/data/cosmos/azcosmos_emulator_tests
+
+go 1.13
```
Comment (seankane-msft) [inline sdk/data/cosmos/azcosmos_emulator_tests/go.mod:3]: Even though this is not going to be publicly released, this should also be go 1.16

Comment (ealsur) [reply sdk/data/cosmos/azcosmos_emulator_tests/go.mod]: Done

File: sdk/data/cosmos/azcosmos/go.mod (hunk 1 above)
Comment (ealsur) [reply sdk/data/cosmos/azcosmos/go.mod]: Done

**Comment by benbp** [comment]
> @benbp We currently have 2 packages, one for black box testing `azcosmos_emulator_tests` I wanted to avoid dumping both inside `data`. @jhendrixMSFT Could you confirm?

I'm not sure what the best solution is here (i.e. if nesting go modules is to be avoided), but I think the convention is to have package names like `data/<track2 service name`, like there already is for `data/aztables`.

**Comment by ealsur** [comment]
> I'm not sure what the best solution is here (i.e. if nesting go modules is to be avoided), but I think the convention is to have package names like `data/<track2 service name`, like there already is for `data/aztables`.

The problem with nesting is that `go test` would execute everything, both the UTs in the root package (azcosmos) and then the nested one (azcosmos_emulator_tests), and it would be fine except for the fact that the Cosmos DB emulator runs only on windows. So it would break in any Linux run. That's why in our YML we run them separately and Emulator tests run only on Windows.

Not sure how to go around this one.

**Comment by weshaggard** [comment]
@ealsur is there no way to condition based on platform or some flag you can pass to go test? If so we should try doing that.

**Comment by ealsur** [comment]
> @ealsur is there no way to condition based on platform or some flag you can pass to go test? If so we should try doing that.

We already discussed that and tried it, `go test` allows for flags to be passed yes but they are compilation flags. The go compilation flags make it so the development process is very hard. The files with the flags do not get compiled in any IDE and it disables any IDE Intellisense or autocompletion.

**Comment by ealsur** [comment]
@weshaggard @benbp - Moved all the files into `sdk/data/azcosmos`, I'll see how to deal with Windows Emulator tests from the dev side

**Review by weshaggard — APPROVED** [review]

**Pull request merged**